game/src/test/scala/io/aigar/controller/LeaderboardControllerSpec.scala

```diff
@@ -1,14 +1,29 @@
 import io.aigar.controller._
+import io.aigar.controller.response._
+
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
 
 import org.scalatra.test.specs2._
+import org.specs2.matcher._
 
 class LeaderboardControllerSpec extends MutableScalatraSpec {
+  protected implicit val jsonFormats: Formats = DefaultFormats
   addServlet(classOf[LeaderboardController], "/*")
 
   "GET / on LeaderboardController" should {
-    "return status 200" in {
+    "return the right data format" in {
       get("/") {
         status must_== 200
+
+        val entries = parse(body).extract[LeaderboardResponse].data
+        entries.foreach(entry => {
+                          entry.team_id must be_>=(0)
+                          entry.score must be_>=(0)
+                        })
+
+        val longerSize = entries.groupBy(_.team_id).foldLeft(0)(_ max _._2.size)
```

Comment:
Took me a bit to understand what this is checking. It is making sure that the team IDs are unique, correct? What if we did something like the following to make it more explicit:
```
// verify that the team IDs are unique
val team_ids = entries.map(_.id)
team_ids.distinct.size == team_ids.size
```